modules/tensor_mechanics/test/tests/2D_slipweakening/tpv2052D.i

```diff
@@ -0,0 +1,257 @@
+  []
+  [tangent_jump_rate]
+    type = TimeDerivativeAux
+    variable = tangent_jump_rate
+    functor = tangent_jump
+    execute_on = 'TIMESTEP_BEGIN'
+  []
+  [Residual_x]
+    type = ProjectionAux
+    variable = resid_slipweakening_x
+    v = resid_x
+    execute_on = 'TIMESTEP_BEGIN'
+  []
+  [Residual_y]
+    type = ProjectionAux
+    variable = resid_slipweakening_y
+    v = resid_y
+    execute_on = 'TIMESTEP_BEGIN'
+  []
+  [restore_x]
+    type = TagVectorAux
+    vector_tag = 'restore_tag'
+    v = 'disp_x'
+    variable = 'resid_x'
+    execute_on = 'TIMESTEP_END'
+  []
+  [restore_y]
+    type = TagVectorAux
+    vector_tag = 'restore_tag'
+    v = 'disp_y'
+    variable = 'resid_y'
+    execute_on = 'TIMESTEP_END'
+  []
+  [StaticFricCoeff]
+    type = FunctionAux
+    variable = mu_s
+    function = func_static_friction_coeff_mus
+    execute_on = 'LINEAR TIMESTEP_BEGIN'
+  []
+  [StrikeShearStress]
+    type = FunctionAux
+    variable = ini_shear_stress
+    function = func_initial_strike_shear_stress
+    execute_on = 'LINEAR TIMESTEP_BEGIN'
+  []
+[]
+
+[Kernels]
+  [./inertia_x]
+    type = InertialForce
+    use_displaced_mesh = false
+    variable = disp_x
+  []
+  [./inertia_y]
+    type = InertialForce
+    use_displaced_mesh = false
+    variable = disp_y
+  []
+  [./Reactionx]
+    type = StiffPropDamping
+    variable = 'disp_x'
+    component = '0'
+  []
+  [./Reactiony]
+    type = StiffPropDamping
+    variable = 'disp_y'
+    component = '1'
+  []
+[]
+
+[Materials]
+  [elasticity]
+      type = ComputeIsotropicElasticityTensor
+      lambda = 32.04e9
+      shear_modulus = 32.04e9
+      use_displaced_mesh = false
+  []
+  [stress]
+      type = ComputeLinearElasticStress
+  []
+  [density]
+      type = GenericConstantMaterial
+      prop_names = density
+      prop_values = 2670
+  []
+  [./czm_mat]
+      type = SlipWeakeningFriction2d
+      disp_slipweakening_x     = disp_slipweakening_x
+      disp_slipweakening_y     = disp_slipweakening_y
+      reaction_slipweakening_x = resid_slipweakening_x
+      reaction_slipweakening_y = resid_slipweakening_y
+      mu_s = mu_s
+      ini_shear_sts = ini_shear_stress
+      boundary = 'Block0_Block1'
+  [../]
+[]
+
+[Functions]
+  [func_static_friction_coeff_mus]
+    type = PiecewiseConstant
+    axis=x
+    x = '-1000e3 -15e3 15e3'
+    y = '10000 0.677 10000.0'
+    direction = left
+  []
+  [func_initial_strike_shear_stress]
+    type = PiecewiseConstant
+    axis=x
+    x = '-1000e3 -9.0e3 -6.0e3 -1.5e3  1.5e3  6.0e3  9.0e3'
+    y = ' 70.0e6 78.0e6 70.0e6 81.6e6 70.0e6 62.0e6 70.0e6'
+  []
+[]
+
+[UserObjects]
+  [recompute_residual_tag]
+      type = ResidualEvaluationUserObject
+      vector_tag = 'restore_tag'
+      force_preaux = true
+      execute_on = 'TIMESTEP_END'
+  []
+[]
+
+[Executioner]
+  type = Transient
+  dt = 0.0025
+  num_steps = 10
+  [TimeIntegrator]
+    type = CentralDifference
+    solve_type = lumped
+  []
+[]
+
+[Outputs]
```

Comment:
Is there csv output you could show instead of exodus that would show the traction separation curve?